### PR TITLE
Improved `Checkpoint` to use score_name as metric's key

### LINKED ISSUE #2129: Use `score_name` as metric's key in Checkpoint
## 🚀 Feature

We can provide a simplification of our code when saving the best model by metric. 

Currently, to save 2 best models according to the validation accuracy, we are doing:
```python
best_model_handler = Checkpoint(
    {"model": model},
    save_handler,
    filename_prefix="best",
    n_saved=2,
    score_name="accuracy",
    score_function=Checkpoint.get_default_score_fn("accuracy"),
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```

However, we can fetch define `score_function` as `Checkpoint.get_default_score_fn("accuracy")` by default if `score_name` is set. Thus, we can remove one line of code:
```python
best_model_handler = Checkpoint(
    {"model": model},
    save_handler,
    filename_prefix="best",
    n_saved=2,
    score_name="accuracy",
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #2129 

Improved `Checkpoint` to be able to fetch score_function as Checkpoint.get_default_score_fn("accuracy") by default if score_name is set, and added a test to cover that.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -84,7 +84,8 @@ class Checkpoint(Serializable):
             :class:`~ignite.engine.engine.Engine` object, and returning a score (`float`). Objects with highest scores
             will be retained.
         score_name: If ``score_function`` not None, it is possible to store its value using
-            ``score_name``. See Notes for more details.
+            ``score_name``. If ``score_function`` is None, can be used alone to fetch ``score_function``.
+            See Notes for more details.
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:88]: Without sending user to Notes below, we can just say that if `score_function is None`, we are using `Checkpoint.get_default_score_fn(score_name)` as default.

File: tests/ignite/handlers/test_checkpoint.py [hunk 2]
```diff
@@ -506,8 +540,8 @@ def test_model_checkpoint_args_validation(dirname):
     with pytest.raises(ValueError, match=r"Directory path '\S+' is not found"):
         ModelCheckpoint(os.path.join(dirname, "non_existing_dir"), _PREFIX, create_dir=False)
 
-    with pytest.raises(ValueError, match=r"If `score_name` is provided, then `score_function` "):
-        ModelCheckpoint(existing, _PREFIX, create_dir=False, score_name="test")
+    # with pytest.raises(ValueError, match=r"If `score_name` is provided, then `score_function` "):
+    #     ModelCheckpoint(existing, _PREFIX, create_dir=False, score_name="test")
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_checkpoint.py:544]: Please remove commented code

Comment (KickItLikeShika) [reply tests/ignite/handlers/test_checkpoint.py:544]: Done.

File: ignite/handlers/checkpoint.py [hunk 3]
```diff
@@ -84,7 +84,8 @@ class Checkpoint(Serializable):
             :class:`~ignite.engine.engine.Engine` object, and returning a score (`float`). Objects with highest scores
             will be retained.
         score_name: If ``score_function`` not None, it is possible to store its value using
-            ``score_name``. See Notes for more details.
+            ``score_name``. If ``score_function`` is None, ``score_name`` can be used alone to fetch ``score_function``
+            via ``Checkpoint.get_default_score_fn(score_name)`` by default.
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:88]:
```suggestion
            ``score_name``. If ``score_function`` is None, ``score_name`` can be used alone to define ``score_function``
            as ``Checkpoint.get_default_score_fn(score_name)`` by default.
```

File: ignite/handlers/checkpoint.py [hunk 4]
```diff
@@ -248,6 +249,10 @@ class Checkpoint(Serializable):
 
         - Checkpoint can save model with same filename.
         - Added ``greater_or_equal`` argument.
+
+    .. versionchanged:: 0.4.6
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:253]:
```suggestion
    .. versionchanged:: 0.5.0
```

File: ignite/handlers/checkpoint.py [hunk 5]
```diff
@@ -248,6 +249,10 @@ class Checkpoint(Serializable):
...
+
+        - `score_name` can be used to fetch `score_function` automatically without providing `score_function`.
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:255]:
```suggestion
        - `score_name` can be used to define `score_function` automatically without providing `score_function`.
```

File: ignite/handlers/checkpoint.py [hunk 6]
```diff
@@ -248,6 +249,10 @@ class Checkpoint(Serializable):
 
         - Checkpoint can save model with same filename.
         - Added ``greater_or_equal`` argument.
+
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:252]: @KickItLikeShika can you update above example: lines 233-241 with the new API
```python
            score_function = Checkpoint.get_default_score_fn("accuracy")

            to_save = {'model': model}
            handler = Checkpoint(
                to_save, DiskSaver('/tmp/models', create_dir=True),
                n_saved=2, filename_prefix='best',
                score_function=score_function, score_name="val_acc",
                global_step_transform=global_step_from_engine(trainer)
)
```

File: tests/ignite/handlers/test_checkpoint.py [hunk 7]
```diff
@@ -274,6 +269,45 @@ def _test(to_save, obj, name):
     _test(to_save, {"model": model.state_dict(), "optimizer": optimizer.state_dict()}, "checkpoint")
 
 
+def test_checkpoint_with_score_name_only():
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_checkpoint.py:272]: Can you put this test into test_checkpoint_with_score_name_and_function as a special case of function as None

Comment (KickItLikeShika) [reply tests/ignite/handlers/test_checkpoint.py:272]: To get the score_function using score_name that requires the metric to be attached, that will not be a good change in this test, so I suggest creating two separate tests for these 2 cases

File: ignite/handlers/checkpoint.py [hunk 8]
```diff
@@ -229,13 +230,11 @@ class Checkpoint(Serializable):
...
             # ...
 
-            score_function = Checkpoint.get_default_score_fn("accuracy")
-
             to_save = {'model': model}
             handler = Checkpoint(
                 to_save, DiskSaver('/tmp/models', create_dir=True),
                 n_saved=2, filename_prefix='best',
-                score_function=score_function, score_name="val_acc",
+                score_name="accuracy",
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:237]: @KickItLikeShika please update this line as well: https://github.com/pytorch/ignite/pull/2146/files#diff-26b2fc842f0567d57c38661f9ef773fe32de32caa2b5e25e9043fcb91238774dR244
```
> ["best_model_9_val_acc=0.77.pt", "best_model_10_val_acc=0.78.pt", ]
```

We have to explain above that evaluator.state.metrics will contain "accuracy" metric which will be used here

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @KickItLikeShika
